js/src/builtin/streams/ReadableStreamInternals.cpp

```diff
@@ -442,12 +480,35 @@ uint32_t js::ReadableStreamGetNumReadRequests(ReadableStream* stream) {
     return 0;
   }
 
-  return reader->requests()->length();
+  ListObject* requests = reader->requests();
+  if (!requests) {
+    return 0;
+  }
+  return requests->length();
 }
 
-// Streams spec, 3.5.11. ReadableStreamHasBYOBReader ( stream )
-//
-// Not implemented.
+/**
+ * Streams spec, 3.5.11. ReadableStreamHasBYOBReader ( stream )
+ */
+[[nodiscard]] bool js::ReadableStreamHasBYOBReader(
+    JSContext* cx, Handle<ReadableStream*> unwrappedStream, bool* result) {
```

Comment:
What should we do if unwrappedStream contains a nullptr? If that is not expected to ever occur, we could add a `MOZ_ASSERT(unwrappedStream)`